packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -274,59 +332,60 @@ export class ProjectLifecycleManager {
 
     this._projectRoot = projectRoot
     this._initializedProject = undefined
-    this._pendingMigrationInitialize = pDefer()
-    this.legacyPluginGuard()
-    Promise.resolve(this.ctx.browser.machineBrowsers()).catch(this.onLoadError)
-    this.verifyProjectRoot(projectRoot)
-    const packageManagerUsed = this.getPackageManagerUsed(projectRoot)
 
     this.resetInternalState()
+
+    this._configManager = this.createConfigManager()
+
+    // Preemptively load these so that they are available when we need them later
+    this.ctx.browser.machineBrowsers().catch(this.onLoadError)
+
+    const packageManagerUsed = this.getPackageManagerUsed(projectRoot)
+
     this.ctx.update((s) => {
       s.currentProject = projectRoot
       s.packageManager = packageManagerUsed
     })
 
+    if (this.readyToInitialize(this._projectRoot)) {
+      this._configManager.initializeConfig().catch(this.onLoadError)
+    }
+  }
+
+  readyToInitialize (projectRoot: string): boolean {
```

Comment:
Would you please add a comment on this function to say what it checks?

Reply:
Added here: [`6bace98` (#20775)](https://github.com/cypress-io/cypress/pull/20775/commits/6bace98a24172d9dcaa72b133eee180e7edba875)